poppy/poppy_core.py

```diff
@@ -1315,88 +1304,98 @@ def calcPSF(self, wavelength=1e-6, weight=None,
         if _USE_FFTW:
             utils.fftw_load_wisdom()
 
-
-        if conf.use_multiprocessing and len(wavelength) > 1 : ######### Parallellized computation ############
+        if conf.use_multiprocessing and len(wavelength) > 1: ######### Parallellized computation ############
             if _USE_FFTW: 
                 _log.warn('IMPORTANT WARNING: Python multiprocessing and fftw3 do not appear to play well together. This may crash intermittently')
                 _log.warn('   We suggest you set   poppy.conf.use_fftw to False   if you want to use multiprocessing().')
             if display:
                 _log.warn('Display during calculations is not supported for multiprocessing mode. Please set poppy.conf.use_multiprocessing.set(False) if you want to use display=True.')
-                _log.warn('For now, display is being set to False.')
-                display=False
+                _log.warn('(Plot the returned PSF with poppy.utils.display_PSF.)')
 
             if save_intermediates:
-                raise NotImplementedError("Can't save intermediate steps if using parallelized code")
-                save_intermediates = False
-                self.intermediate_wfs = []
+                _log.warn('Saving intermediate steps does not take advantage of multiprocess parallelism. '
+                          'Set save_intermediates=False for improved speed.')
 
             # do *NOT* just blindly try to create as many processes as one has CPUs, or one per wavelength either
             # This is a memory-intensive task so that can end up swapping to disk and thrashing IO
-            nproc = conf.n_processes if conf.n_processes > 1 else utils.estimate_optimal_nprocesses(self, nwavelengths=len(wavelength))
-
-            pool = multiprocessing.Pool( int(nproc) )  # be sure to cast to int, will fail if given a float even if of integer value
+            nproc = conf.n_processes if conf.n_processes > 1 \
+                                     else utils.estimate_optimal_nprocesses(self, nwavelengths=len(wavelength))
 
+            # be sure to cast to int, will fail if given a float even if of integer value
+            pool = multiprocessing.Pool(int(nproc))
 
             # build a single iterable containing the required function arguments
             _log.info("Beginning multiprocessor job using {0} processes".format(nproc))
-            iterable = [(self, wavelen, wt, kwargs, _USE_FFTW) for wavelen, wt in zip(wavelength, normwts)]
-            results = pool.map(_wrap_propagate_for_multiprocessing, iterable)
+            worker_arguments = [(self, wlen, retain_intermediates, normalize, _USE_FFTW)
+                                for wlen in wavelength]
+            results = pool.map(_wrap_propagate_for_multiprocessing, worker_arguments)
```

Comment:
Since all the wavelengths and their intermediate wavefronts are accumulated here before any combining is done, this could potentially get very memory intensive for large numbers of wavelengths, many intermediate planes, etc. 

That's not necessarily a problem - and I don't see any easy work around - but we should make a note somewhere in the documentation, to the effect that users should be aware of this and should consider using single process mode if they run into memory limitations.